File: personas.js
const readlineSync = require("readline-sync");

let personas = []

function ingresarPersonas(){
    let nombre = readlineSync.question("Ingrese el nombre: ");
    let apellido = readlineSync.question("Ingrese el apellido: ");
    let dni = readlineSync.question("Ingrese el DNI: ");
    let telefonos = readlineSync.question("Ingrese los telefonos separados por coma: ").split(",");
    let hijos = readlineSync.question("Ingrese los nombres de los hijos separados por coma: ").split(",");

    personas.push([nombre, apellido, dni, telefonos, hijos])
}

function mostrarDatos(){
    console.log("\n Datos de personas:");
    personas.forEach((persona) =>{
        console.log(`Nombre: ${persona[0]} ${persona[1]}, DNI: ${persona[2]}`);
        console.log(`Telefono: ${persona[3].join(", ")}`);
        console.log(`Hijos: ${persona[4].join(", ")}`);
        console.log(`Cantidad de hijos: ${persona[4].length}, Cantidad de telefonos: ${persona[3].length}\n`);
    })
}

function filtrarDni(){
    let dni = readlineSync.question("Ingrese el DNI de la persona que quiera encontrar:");
    let personaEncontrada = personas.find(persona => persona[2] === dni);
    if(personaEncontrada){
        console.log(`Nombre: ${personaEncontrada[0]} ${personaEncontrada[1]}, DNI: ${personaEncontrada[2]}`);
        console.log(`Telefono: ${personaEncontrada[3].join(", ")}`);
        console.log(`Hijos: ${personaEncontrada[4].join(", ")}`);
        console.log(`Cantidad de hijos: ${personaEncontrada[4].length}, Cantidad de telefonos: ${personaEncontrada[3].length}\n`);
    } else{
        console.log("Persona no encontrada");
    }
}

function mostrarMenu(){
    let opcion;
    do{
        console.log("\nMenu:");
        console.log("1.Ingrese nueva persona");
        console.log("2.Mostrar todos los datos");
        console.log("3.Filtrar por DNI");
        console.log("4.Salir")
        opcion = readlineSync.question("Ingrese la opcion: ");

        switch(opcion){
            case "1":
                ingresarPersonas();
                break;
            case "2":
                mostrarDatos();
                break;
            case "3":
                filtrarDni();
                break;
            case "4":
                console.log("Programa finalizado");
                break;
            default:
                console.log("Opcion no valida");
        }
    } while (opcion !== "4");
}

mostrarMenu();

File: personas.py
def ingresar_persona():
    nombre = input("Ingrese el nombre: ")
    apellido = input("Ingrese el apellido: ")
    dni = input("Ingrese el DNI: ")
    telefonos = input("Ingrese los télefonos separados por coma:").split(", ")
    hijos = input("Ingrese los nombres de los hijos separados por coma:").split(", ")

    personas.append([nombre, apellido, dni, telefonos, hijos])

def mostrar_datos():
    print("\nDatos de personas:")
    for persona in personas:
        print(f"Nombre: {persona[0]} {persona[1]}, DNI: {persona[2]}")
        print(f"Telefonos: {', '.join(persona[3])}")
        print(f"Hijos: {', '.join(persona[4])}")
        print(f"Cantidad de hijos: {len(persona[4])}, Cantidad de telefonos: {len(persona[3])}\n")

def filtrar_dni():
    dni = input("Ingrese el DNI de la persona que quiera encontrar:")
    persona_encontrada = next((persona for persona in personas if persona[2] == dni), None)
    if persona_encontrada:
        print(f"Nombre: {persona_encontrada[0]} {persona_encontrada[1]}, DNI: {persona_encontrada[2]}")
        print(f"Telefonos: {', '.join(persona_encontrada[3])}")
        print(f"Hijos: {', '.join(persona_encontrada[4])}")
        print(f"Cantidad de hijos: {len(persona_encontrada[4])}, Cantidad de telefonos: {len(persona_encontrada[3])}\n")
    else:
        print("Persona no encontrada")

def mostrar_menu():
    while True:
        print("\nMenu:")
        print("1.Ingrese nueva persona")
        print("2.Mostrar todos los datos")
        print("3.Filtrar por DNI")
        print("4.Salir")
        opcion = input("Ingrese la opcion: ")

        if opcion == "1":
            ingresar_persona()
        elif opcion == "2":
            mostrar_datos()
        elif opcion == "3":
            filtrar_dni()
        elif opcion == "4":
            print("Programa finalizado")
            break
        else:
            print("Opcion no valida")

personas = []
mostrar_menu()